code/modules/paperwork/stamps.dm

```diff
@@ -12,6 +12,61 @@
 	item_color = "cargo"
 	pressure_resistance = 2
 	attack_verb = list("stamped")
+	var/stamp_color = "#339900"		//used for stamp marks, based on the item sprites.. not the on paper images(those are very bland)
+
+/obj/item/stamp/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if(!istype(target, /mob/living/carbon/human) || !istype(user, /mob/living/carbon/human) || !proximity)
+		return
+
+	var/mob/living/carbon/human/H = target
+	var/list/paramslist = list()
+	var/attackedSide				//the stamp mark will appear on this side, NORTH = forwards, SOUTH = back, EAST = right, WEST = left
+	paramslist = params2list(params)
+	var/xOffset = text2num(paramslist["icon-x"]) - 16
+	var/yOffset = text2num(paramslist["icon-y"]) - 16
+
+	switch(H.dir)					//This makes the attackSide be the side that currently faces the 'camera'
+		if(NORTH)
+			attackedSide = SOUTH
+		if(SOUTH)
+			attackedSide = NORTH
+		if(EAST)
+			attackedSide = EAST
+		if(WEST)
+			attackedSide = WEST
+
+	var/icon/new_stamp_mark = icon('icons/effects/stamp_marks.dmi', "stamp[rand(1,3)]_[attackedSide]")
+	new_stamp_mark.Shift(EAST, xOffset)
+	new_stamp_mark.Shift(NORTH, yOffset)
+
+	var/mutable_appearance/targetBaseIcon = mutable_appearance()			//we will use this so the stamps only appear on the human's body and not
+	targetBaseIcon.dir = H.dir												//on their backpack or items they hold in their hand
+	targetBaseIcon.overlays += H.overlays_standing[BODY_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[LIMBS_LAYER]				//Currently this is here and not in update_icons because my only known
+	targetBaseIcon.overlays += H.overlays_standing[UNDERWEAR_LAYER]			//method of turning an image into an icon destroys its directionality
+	targetBaseIcon.overlays += H.overlays_standing[HAIR_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[UNIFORM_LAYER]			//some species' bodies dont take up all the space their uniforms do
+	new_stamp_mark.Blend(getFlatIcon(targetBaseIcon), BLEND_MULTIPLY)		//cut out any parts of the stamp mark that aren't on base human
+
+	var/image/stamp_image = image(new_stamp_mark)
+	stamp_image.text = icon_state
```

Comment:
Instead of setting this to icon state and then have a giant switch statement later on for human examines, I'm going to suggest a bit of a different approach.

Create a static associative list for stamps, that stores the icon_state to examine text, i.e:
```dm
var/static/list/stamp_examine_list = list(
	"stamp-qm" = "Quartermaster approved"
	"stamp-law" = "Justice Department approved"
	// and so on
)
```
Then here you can do:
```dm
stamp_image.text = stamp_examine_list[icon_state]
```
If icon state is `"stamp-qm"` for example, `stamp_image.text` will be set to `"Quartermaster approved"`. Instead of doing the switch in the human examine proc, you can just do `msg += stamp_image.text`

Reply:
ah so this is what a associative list is, neat stuff